tools/CI/check-gitrange.bash

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2018 Intel Corporation. All rights reserved.
+
+set -e
+
+usage()
+{
+    cat <<EOFUSAGE
+Sample usage:
+
+    $0 origin/master... text/x-shellscript   shellcheck  -x -f gcc
+
+    $0 origin/master... text/x-python        pylint  --disable=C
+
+EOFUSAGE
+
+    exit 1
+}
+
+main()
+{
+    # git outputs relative paths
+    local git_top; git_top=$(git rev-parse --show-toplevel)
+    cd "$git_top"
+
+    local diffrange="$1"; shift || usage
+
+    printf '%s checking diff range: %s\n\n' "$0" "$diffrange"
+
+    local checked_ftype="$1"; shift || usage
+    file --list | grep -qF "$checked_ftype" ||
+        die 'The file command does not know what %s is\n' "$checked_ftype"
+
+    local checker="$1"; shift || usage
+    type "$checker" || die 'Checker %s not found\n' "$checker"
+
+    # Triple dot "git log A...B" includes commits not relevant to triple
+    # dot "git diff A...B"
+    local logrange=${diffrange/.../..}
```

Comment:
That was a complex path for this three dot. Can we move this into travis.yml and keep the what is same here? I think put three dot as argument my need some explanation. This did not follow user experience with `git log`

Reply:
Travis provides only a diffrange syntax (triple dots)

This script needs both the logrange syntax (double dots) and the diffrange syntax (triple dots). 

If a conversion is added in `.travis.yml` then:
- it will have to be done on every travis line
- the reverse conversion will have to be done in this script anyway.

This means many more conversions, more tedious and error-prone than just only one conversion in only one place.